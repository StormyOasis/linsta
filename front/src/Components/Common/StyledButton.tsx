import React from "react";
import { styled } from "styled-components";

type StyledButtonProps = {
    name?: string;
    onClick?: any;
    type?: "button" | "submit" | "reset" | undefined;
    disabled?: boolean;
    text: string;
    style?: any;
    datatestid?: string;
    useSecondaryColors?: boolean;
};

const StyledButtonComponent = styled.button<{ $props?: any, $useSecondaryColors?: boolean | undefined }>`
    border: none;
    border-radius: 8px;
    text-decoration: none;
    align-items: center;
    font-weight: 600;
    justify-content: center;
    text-wrap: nowrap;
    color: ${props => props.$useSecondaryColors ? props.theme['colors'].buttonSecondaryTextColorDefault : props.theme['colors'].buttonTextColorDefault};
    background-color: ${(props) => props.disabled ?
        (props.$useSecondaryColors ? props.theme['colors'].buttonSecondaryDefaultColorTrans : props.theme['colors'].buttonDefaultColorTrans) :
        (props.$useSecondaryColors ? props.theme['colors'].buttonDefaultSecondaryColor : props.theme['colors'].buttonDefaultColor)};
    display: flex;
    height: 34px;
    position: relative;
    text-align: center;
    padding-left: 16px;
    padding-right: 16px;
    flex-direction: row;
    margin-bottom: 12px;
    margin-top: 0px;
    margin-left: 40px;
    margin-right: 40px;

    cursor: ${(props) => (props.disabled ? "default" : "pointer")};

    &:hover {
    background-color: ${(props) =>
        props.disabled ?
            (props.$useSecondaryColors ? props.theme['colors'].buttonSecondaryDefaultColorTrans : props.theme['colors'].buttonDefaultColorTrans) :
            (props.$useSecondaryColors ? props.theme['colors'].buttonSecondaryOnHoverColor : props.theme['colors'].buttonOnHoverColor)};
    }
`;

const StyledButton: React.FC<StyledButtonProps> = (props: StyledButtonProps) => {
    return (
        <StyledButtonComponent
            type={props.type}
            name={props.name}
            disabled={props.disabled}
            onClick={props.onClick}
            style={props.style}
            $useSecondaryColors={props.useSecondaryColors}
            data-testid={props.datatestid}>
            {props.text}
        </StyledButtonComponent>
    );
}

export default StyledButton;