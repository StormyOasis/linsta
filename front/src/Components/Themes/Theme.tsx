import React from "react";
import { ThemeProvider } from "styled-components";

const defaultTheme = {
    colors: {
        defaultLinkColor: "black",
        defaultTextColor: "black",
        backgroundColor: "white",
        backgroundDisabled: "rgba(0, 0, 0, .6)",
        borderDefaultColor: "rgb(220, 220, 220)",
        borderDarkColor: "rgb(180,180, 180)",
        inputBackgroundColor: "rgb(250, 250, 250)",
        navLinkHoverColor: "rgb(236, 236, 236)",
        buttonTextColorDefault: "white",
        buttonSecondaryTextColorDefault: "black",
        buttonDefaultColor: "rgb(0, 150, 245)",
        buttonDefaultSecondaryColor: "rgb(240, 240, 240)",
        buttonDefaultColorTrans: "rgba(0, 150, 245, .5)",
        buttonSecondaryDefaultColorTrans: "rgba(240, 240, 240, .5)",
        buttonOnHoverColor: "rgb(25, 120, 240)",        
        buttonSecondaryOnHoverColor: "rgb(220, 220, 220)",        
        inputTextColor: "rgb(115,115,115)",
        navLinkTextColor: "rgb(0, 50, 100)",
        cropperAspectBkgnd: "rgba(25,25,25,.6)",
        cropperAspectBkgndNoTrans: "rgb(25,25,25)",
        mediumTextColor: "rgb(120, 120, 120)",  
        mediaSliderButtonColor: "rgba(0,0,0, .6)",
        mediaSliderButtonBkgndColor: "rgba(225,225,225, .4)",
        warningTextColor: "rgb(235, 70, 90)"
    },    

    input: {
        backgroundColor: "#EFEFEF",        
    },

    sizes: {
        sideBarNavWidthDefault: "244px",
        sideBarNavWidthNarrow: "72px",
        sideBarNavBottomHeight: "50px",
        cropperHeight: "275px",
        defaultModalWidth: "600px",
        maxModalWidth: "75vw",
        maxModalHeight: "75vh",
        minModalHeight: "300px",
        minPostTextEditorHeight: "140px",
        maxPostTextEditorHeight: "140px",
        feedPostMinWidth: "400px",
        maxCommentModalContentHeight: "75vh",
        minCommentModalContentHeight: "300px",
    },

    breakpoints: {
        sm: "576",
        md: "768",
        lg: "1280"
    }
};

export const LexicalEditorTheme = {
    ltr: 'ltr',
    rtl: 'rtl',
};

const Theme = ({children}: any) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
};

export default Theme;
