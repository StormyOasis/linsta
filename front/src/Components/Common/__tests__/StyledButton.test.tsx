import React from 'react'
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom';
import Jest from 'jest';
import StyledButton from '../StyledButton';
import Theme from '../../Themes/Theme';
import { MemoryRouter } from 'react-router-dom';


test("StyledButton Component", () => {
    const r = render(
        <MemoryRouter>
            <Theme>
                <StyledButton type="button" onClick={() => true} text="Hello"></StyledButton>
            </Theme>
        </MemoryRouter>).asFragment();

    expect(r).toMatchSnapshot();
});

test("StyledButton Component With Style Override", () => {
    const styleOverride = { color: "red" };
    const r = render(
        <MemoryRouter>
            <Theme>
                <StyledButton type="submit" text="Hello" onClick={() => true} style={styleOverride}></StyledButton>
            </Theme>
        </MemoryRouter>).asFragment();

    expect(r).toMatchSnapshot();
})