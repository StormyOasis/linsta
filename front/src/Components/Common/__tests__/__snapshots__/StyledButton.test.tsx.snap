// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component 1`] = `
<DocumentFragment>
  <button
    class="StyledButton__StyledButtonComponent-sc-8t4kac-0 jZJObc"
    type="button"
  >
    Hello
  </button>
</DocumentFragment>
`;

exports[`StyledButton Component With Style Override 1`] = `
<DocumentFragment>
  <button
    class="StyledButton__StyledButtonComponent-sc-8t4kac-0 jZJObc"
    style="color: red;"
    type="submit"
  >
    Hello
  </button>
</DocumentFragment>
`;
